### PR TITLE
Implemented client.

### PR DESCRIPTION
**Pull request by mwhittaker**
The client pings the server every `raft::CLIENT_PING_PERIOD` milliseconds. Run with `raft client`.

### CODE REVIEW

**Comment by mwhittaker** [comment]
@yuxiaotan, can you look at this too?

File: src/client.rs [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
+use std::io::{TcpStream, Timer};
+
 pub fn main() {
-    println!("client")
+    let mut stream = TcpStream::connect(SERVER_HOST, SERVER_PORT);
+
+    let mut timer = {
+        match Timer::new() {
+            Ok(timer) => timer,
+            Err(e) => {
```
Comment (yuxiaotan) [inline src/client.rs:10]: When would this code be triggered? I am assuming you are matching on something that is newly instantiated each time (Timer::new()).

Comment (mwhittaker) [inline src/client.rs:10]: Are you asking when will `Timer::new()` yield an `Err`?

Comment (yuxiaotan) [inline src/client.rs:10]: Yes

Comment (mwhittaker) [inline src/client.rs:10]: :) I wish I could tell you. All I know is that [`Timer::new()`](http://doc.rust-lang.org/std/io/timer/struct.Timer.html#method.new) returns an `IoResult<Timer>`. Not sure why.

Comment (mwhittaker) [inline src/client.rs:10]: Here's the source for `Timer::new()`. I don't understand it, but it involves errors.

```
/// Creates a new timer which can be used to put the current task to sleep
/// for a number of milliseconds, or to possibly create channels which will
/// get notified after an amount of time has passed.
pub fn new() -> IoResult<Timer> {
    LocalIo::maybe_raise(|io| {
        io.timer_init().map(|t| Timer { obj: t })
    }).map_err(IoError::from_rtio_error)
}
```

Comment (yuxiaotan) [inline src/client.rs:10]: Lol I don't either. But makes sense why you have to deal with it now. Looks great to me. Merged.

**Comment by mwhittaker** [comment]
@yuxiaotan, :+1:

**Comment by ambarsoni** [comment]
@mwhittaker I finally got around to this and it looks like a logical next step!

**Comment by mwhittaker** [comment]
Thank you @ambarsoni!

**Pull request merged**